v2/apiserver/internal/authx/rest/sessions_endpoints.go

```diff
@@ -0,0 +1,132 @@
+package rest
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/brigadecore/brigade/v2/apiserver/internal/authx"
+	"github.com/brigadecore/brigade/v2/apiserver/internal/lib/restmachinery"
+	"github.com/brigadecore/brigade/v2/apiserver/internal/meta"
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+)
+
+// SessionsEndpoints implements restmachinery.Endpoints to provide
+// Session-related URL --> action mappings to a restmachinery.Server.
+type SessionsEndpoints struct {
+	AuthFilter restmachinery.Filter
+	Service    authx.SessionsService
+}
+
+// Register is invoked by restmachinery.Server to register Session-related
+// URL --> action mappings to a restmachinery.Server.
+func (s *SessionsEndpoints) Register(router *mux.Router) {
+	// Create session
+	router.HandleFunc(
+		"/v2/sessions",
+		s.create, // No filters applied to this request
+	).Methods(http.MethodPost)
+
+	// Delete session
+	router.HandleFunc(
+		"/v2/session",
+		s.AuthFilter.Decorate(s.delete),
+	).Methods(http.MethodDelete)
+
+	// OIDC callback
+	router.HandleFunc(
+		"/v2/session/auth",
+		s.authenticate, // No filters applied to this request
+	).Methods(http.MethodGet)
+}
+
+func (s *SessionsEndpoints) create(w http.ResponseWriter, r *http.Request) {
+	// nolint: errcheck
+	rootSessionRequest, _ := strconv.ParseBool(r.URL.Query().Get("root"))
```

Comment:
Is there no scenario where this could fail?

Reply:
Arguably, there isn't.

```
// ParseBool returns the boolean value represented by the string.
// It accepts 1, t, T, TRUE, true, True, 0, f, F, FALSE, false, False.
// Any other value returns an error.
func ParseBool(str string) (bool, error) {
	switch str {
	case "1", "t", "T", "true", "TRUE", "True":
		return true, nil
	case "0", "f", "F", "false", "FALSE", "False":
		return false, nil
	}
	return false, syntaxError("ParseBool", str)
}
```

The only way that function call produces an error is if the input is not recognized as a string representation of `true` or `false`, but in that case, `false` is _also_ returned. So the function effectively returns `true` _if and only if_ the input is a recognized string representation of `true` and false for all other cases.

If we agree that anything that isn't "truthy" is false, then this works the way we'd expect without any need to check the error.